e2e/cephfs.go

```diff
@@ -850,6 +850,220 @@ var _ = Describe(cephfsType, func() {
 				}
 			})
 
+			By("Test subvolume snapshot and restored PVC metadata", func() {
+				err := createCephFSSnapshotClass(f)
+				if err != nil {
+					e2elog.Failf("failed to create CephFS snapshotclass: %v", err)
+				}
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC: %v", err)
+				}
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				// create snapshot
+				snap.Name = f.UniqueName
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create snapshot (%s): %v", snap.Name, err)
+				}
+				subvolumes, err := listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolumes: %v", err)
+				}
+				if len(subvolumes) == 0 {
+					e2elog.Failf("cephFS subvolumes list is empty %s", fileSystemName)
+				}
+				snaps, err := listCephFSSnapshots(f, fileSystemName, subvolumes[0].Name, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolume snapshots: %v", err)
+				}
+				if len(snaps) == 0 {
+					e2elog.Failf("cephFS snapshots list is empty %s/%s", fileSystemName, subvolumes[0].Name)
+				}
```

Comment:
This is a good one :+1: . can you please open an issue to add this check for all the tests where we are creating the snapshots. Looks like we are not validating the snapshot counts in existing checks

Reply:
Sure will open one, thanks!

Reply:
@Madhu-1 here is one https://github.com/ceph/ceph-csi/issues/3224